content/docs/rfcs/23/README.md

```diff
@@ -0,0 +1,155 @@
+---
+slug: 23
+title: 23/WAKU2-TOPICS
+name: Waku v2 Topic Usage Recommendations
+status: draft
+editor: Oskar Thoren <oskar@status.im>
+contributors:
+---
+
+This document outlines recommended usage of topics in Waku v2. In [10/WAKU2 spec](/spec/10) there are two types of topics:
+
+- Pubsub topics, used for routing
+- Content topics, used for content-based filtering
+
+## Pubsub topics
+
+Pubsub topics are used for routing of mesages, see [11/WAKU2-RELAY](/spec/11) spec for more details of how this routing works.
+
+As written in [10/WAKU2 spec](/spec/10) there is a default Pubsub topic:
+
+`/waku/2/default-waku/proto`
+
+This indicates that
+
+1) It relates to the Waku problem domain
+2) version is 2
+3) name indicates what is being exchanged, which in this case is WakuMessages over a single topic and
+4) that the data in PubSub field is protobuf (unlike Eth2 where it is `ssz_snappy`) as determined by WakuMessage.
+
+The default PubSub topic SHOULD be used for all protocols.
+However, in certain situations other topics MAY be used.
+
+### Pubsub topic format
+
+PubSub topics SHOULD follow the folloing structure:
+
+`/waku/2/TopicName/Encoding`
+
+This namespaced structure makes things like compatibility and discoverability and automatic handling of new topics easier.
+For example, if the encoding of the payload is changed, compression is introduced, etc.
+
+For more on this format of PubSub topics, see [Ethereum 2 P2P spec](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/phase0/p2p-interface.md#topics-and-messages) where inspiration for this format was taken from.
+
+### Separation of two applications example
+
+Let's say we have some M2M topic that is heavy and experience a lot of heavy traffic and a Status Network one. This can be segregated into:
+
+```
+/waku/2/waku-status/proto
+/waku/2/waku-walletconnect/proto
+```
+
+This indicates that they are WakuMessages but for different domains completely.
+
+### Topic sharding example
+
+Topic sharding is currently not supported by default, but is planned for the future in order to deal with increased network traffic. Here's a sketch for what this might look like:
+
+```
+waku/2/waku-9_shard-0/proto
+...
+waku/2/waku-9_shard-9/proto
+```
+
+This indicates explicitly that the network traffic has been partitioned into 10 buckets.
+
+### Compression example
+
+Not yet implemented, but would be easy to add with:
+
+`/waku/2/default-waku/proto_snappy`
```

Comment:
```suggestion
`/waku/2/default-waku/proto_snappy`
```
Why mixing snake and kebab case? Consistency across the field would make easier to not end up with `proto_zip` and `proto-zip` protocols.

Reply:
proto_snappy is what was used in eth2, this is arbitrary and an example.